client/components/ProtectedRoute.jsx

```diff
@@ -1,19 +1,57 @@
 import { useState, useEffect } from "react";
 import { Navigate } from "react-router-dom";
+import { auth } from "../src/firebase";
 
 export default function ProtectedRoute({ children }) {
-  const [loading, setLoading] = useState(true);
+  const [loading, setLoading]      = useState(true);
   const [isAuthenticated, setAuth] = useState(false);
 
   useEffect(() => {
-    fetch(`${import.meta.env.VITE_API_URL}/auth/me`, {
-      credentials: "include",
-    })
-      .then(res => {
-        if (res.ok) setAuth(true);
-      })
-      .catch(() => setAuth(false))
-      .finally(() => setLoading(false));
+    let cancelled = false;
```

Comment:
```suggestion
    let cancelled = false;
    const controller = new AbortController();
    const signal = controller.signal;
```
You can use AbortController to cancel fetch requests on cleanup